root/edit/components/EditNote.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {CatalystContext} from '../../context';
+import EditorLink from '../../static/scripts/common/components/EditorLink';
+import expand2react from '../../static/scripts/common/i18n/expand2react';
+import formatUserDate from '../../utility/formatUserDate';
+
+import EditorTypeInfo from './EditorTypeInfo';
+
+type Props = {
+  +edit: EditT,
+  +editNote: EditNoteT,
+  +index: number,
+};
+
+function returnNoteAnchor(edit, index) {
+  return `note-${edit.id}-${index + 1}`;
+}
+
+function returnVoteClass(vote, isOwner) {
+  let className = '';
+
+  if (vote) {
+    className = vote.vote_name;
+  }
+
+  if (isOwner) {
+    if (vote) {
+      className += ' ';
+    }
+    className += 'owner';
+  }
+
+  return className;
+}
+
+const EditNote = ({
+  edit,
+  editNote,
+  index,
+}: Props): React.Element<'div'> => {
+  const $c = React.useContext(CatalystContext);
+  const isModBot = editNote.editor_id === 4;
+  const anchor = returnNoteAnchor(edit, index);
+  const isOwner = edit.editor_id === editNote.editor_id;
+  const lastRelevantVote = edit.votes.find(vote => (
+    vote.editor_id === editNote.editor_id &&
+    !vote.superseded
+  ));
+  const showVotingIcon = lastRelevantVote && (
+    lastRelevantVote.vote_name === 'Approve' ||
+    lastRelevantVote.vote_name === 'No' ||
+    lastRelevantVote.vote_name === 'Yes'
```

Comment:
We should probably import the `EDIT_VOTE_*` constants and compare those to `.vote`, since we might eventually stop outputting `vote_name` and calculate that in JS instead.

Reply:
I added a `getVoteName` utility function.